### DISCUSSION TITLE
How can I load 2 different graphs with PyG and pytorch_lighting ?

### DISCUSSION THREAD

**Discussion by xiachenrui** [pyg-team/pytorch_geometric#7085] [Q&A] [answered]
Hi all,
    I am trying to do contrastive learning between two different graphs. Now I bulid two `Data` objects, but I found that `LightningNodeData` object can only  receive one `Data` as input. How can I input two graphs at the same time.

Thanks !
Chen-Rui

**Comment by xiachenrui** [comment, marked as answer]
I need to explain that the nodes of the two graphs are shared, but the edges are completely different, so I cannot use `Batch` to merge the two graphs

**Reply by rusty1s** [reply]
In general, `DataLoader` supports returning tuples of `data` objects, so something like
```python
class Dataset(...)
    def __getitem__(self, idx):
         return dataset1[idx], dataset2[idx]
         
dataset = Dataset()
loader = DataLoader(dataset, ...)

for batch1, batch2 in loader:
    pass
```
already works.

You are right that `LightningNodeData` doesn't support this, but you can consider writing your own `DataModule` for PyTorch Lightning to support this.

**Reply by xiachenrui** [reply]
Thanks for your kind reply！
